b3lb/rest/b3lb/lb.py

```diff
@@ -130,18 +147,19 @@ def get_node_params_by_lowest_workload(cluster_group):
         return None
 
 
-def get_slide_body_for_post(slide):
-    slide_path = "rest/slides/{}".format(slide)
-    try:
-        slide_file = open(slide_path, "rb")
-        body = '<modules><module name="presentation"><document name="{}">{}</document></module></modules>'.format(slide, base64.b64encode(slide_file.read()).decode())
-        slide_file.close()
-    except FileNotFoundError:
-        body = None
+def get_slide_body_for_post(asset, secret):
+    if asset.slide_filename:
+        file_name = asset.slide_filename
+    else:
+        file_name = "{}.{}".format(secret.tenant.slug.lower(), asset.slide.filename.split(".")[-1])
+
+    file_url = "{}/b3lb/t/{}/slide".format(settings.B3LP_API_BASE_DOMAIN, secret.tenant.slug.lower())
+
+    body = '<modules><module name="presentation"><document url="{}" filename="{}"></module></modules>'.format(file_url, file_name)
```

Comment:
The filename should be secured against escaping attacks (i.e. a filename may contain quotes).

Reply:
xml_escaping used in model property